### PR TITLE
update_experiment now rejects invalid URLs.

### PR DESCRIPTION
**Pull request by dctaf**
This fixes the current issue where experiment design URLs can be updated to invalid URLs which break downstream validation. We retain special treatment of NULLs to support our ad-hoc "patch" protocol where "None" means "no change", and empty string means "clear".

### CODE REVIEW

File: src/xngin/apiserver/routers/admin/test_admin.py [hunk 1]
```diff
@@ -1447,6 +1447,37 @@ async def test_update_experiment(testing_experiment, ppatch, pget):
...
+
+@pytest.mark.parametrize(
+    ("url", "expected_url"),
+    [
+        ("http://example.com", "http://example.com/"),
+        ("https://example.com", "https://example.com/"),
+        ("https://drive.google.com/...?q=1", "https://drive.google.com/...?q=1"),
+    ],
+)
+async def test_update_experiment_url_valid(testing_experiment, ppatch, pget, url, expected_url):
```
Comment (qixotic) [inline src/xngin/apiserver/routers/admin/test_admin.py:1468]: Can you add one more test in which a known non-empty design url is then updated with a "", resulting in the next fetch of the experiment to have `design_url is None`?
